Firestore/core/src/firebase/firestore/remote/connectivity_monitor.cc

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/remote/connectivity_monitor.h"
+
+#include "Firestore/core/src/firebase/firestore/util/hard_assert.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+void ConnectivityMonitor::SetInitialStatus(NetworkStatus new_status) {
+  HARD_ASSERT(!status_.has_value(),
+              "SetInitialStatus should only be called once");
+  status_ = new_status;
+}
+
+void ConnectivityMonitor::MaybeInvokeCallbacks(NetworkStatus new_status) {
+  // TODO(varconst): cancel the enqueued operation in destructor or use
```

Comment:
It's not the first time the need to cancel enqueued operations come up. I'm a little tempted to abuse `EnqueueWithDelay`, which is cancelable.